### DISCUSSION TITLE
Responsive Buttons?

### DISCUSSION THREAD

**Discussion by Sreenivas1323** [chakra-ui/chakra-ui#2276] [Q&A] [answered]
So, i have been working on a small project for myself, trying Chakra Ui was damn easy with the finest Documentation Out there!
But, as a JavaScript Newbiee, I had this question in mind, like how to make Chakra-buttons responsive!

i have tried this 

` <IconButton  icon={<RiTwitterFill /> } size={{base: "xs", md:"sm", xl:"md"}} /> `

but, no Use, 
Can anyone help me out?

**Comment by ljosberinn** [comment, marked as answer]
On which version are you on? In v1, you need to use `boxSize` instead of size, as mentioned [here](https://next.chakra-ui.com/docs/migration#5-update-layout-size-prop) as a breaking change between v0 and v1.

**Reply by biowaffeln** [reply]
This applies to "Box" and other components, but not to Buttons which still use the size prop, see [here](https://chakra-ui.com/docs/form/button). Is there a way to make that size prop responsive?

**Reply by I-am-abdulazeez** [reply]
This this done yet?

**Reply by jigz** [reply]
if you really wanna use the size prop, try the `useBreakpointValue` hook
```
 const buttonSize = useBreakpointValue(['xs', 'md'])
  <Button size={buttonSize}>Button</Button
````

**Reply by keithburgie-rts** [reply]
Thanks @jigz. Why doesn't this prop use the breakpoints object like everything else?

**Reply by joernroeder** [reply]
I'm wondering as well why this is not the case. any reason?

**Reply by TylerAPfledderer** [reply]
@keithburgie-rts, so in exploring the type signatures, the props that are responsive use the interface `ResponsiveValue` to be able to take an object that assigns values by breakpoint. This Interface is applied via the type `Token` which takes two assertions, `CSSType` and a `ThemeKey` to use from the Chakra theme. The former is a single CSS property.

For example, check out any of the space props: https://github.com/chakra-ui/chakra-ui/blob/main/packages/styled-system/src/config/space.ts
Token type found on line 16 in this file: https://github.com/chakra-ui/chakra-ui/blob/main/packages/styled-system/src/utils/types.ts

So the responsive props are intended to take a value that is applied to a single property, or in the case of a prop like `paddingX`, two related properties that hold the same unit value.

In the case of the prop `size` or the prop `colorScheme`, it is probably not possible because they return objects of different properties and values.

For example, with the `Button` component, the default size `sm` holds this object:
```js
sm: {
  h: 8,
  minW: 8,
  fontSize: "sm",
  px: 3,
},
```
The prop wouldn't be able to accept the type `ResponsiveValue` because of an object like this.

Using `useBreakpointValue` as @jigz works because it returns a single string which is what the `size` prop is looking for.

I would not know how difficult or practical it would be to create a responsive type that would work for these styling objects, as it would not be consistent between the different props that render multiple varying properties, versus using `useBreakpointValue`.

P.S. I welcome anyone that can better clarify any of the above! 😅
